tests/test_mtx.py

```diff
@@ -53,15 +54,31 @@ def test_check_bundle(self):
             MTXIngestor.check_bundle(barcodes_bad, genes, mtx_dimensions)
         expected_msg = (
             f"Expected {expected_barcodes} cells and {expected_genes} genes. "
-            f"Got {len(barcodes_bad)} cells and {len(genes)} genes."
+            f"Got {len(barcodes_bad)} cells and {len(genes)} genes. "
+            f"Please check the files of your mtx bundle (barcode, feature, mtx) for errors."
         )
         self.assertEqual(str(cm.exception), expected_msg)
 
         with self.assertRaises(ValueError) as cm:
             MTXIngestor.check_bundle(barcodes_bad, gene_short, mtx_dimensions)
         expected_msg = (
             f"Expected {expected_barcodes} cells and {expected_genes} genes. "
-            f"Got {len(barcodes_bad)} cells and {len(gene_short)} genes."
+            f"Got {len(barcodes_bad)} cells and {len(gene_short)} genes. "
+            f"Please check the files of your mtx bundle (barcode, feature, mtx) for errors."
+        )
+        self.assertEqual(str(cm.exception), expected_msg)
+
+        genes_as_barcodes = [1, 2, 3, 4, 5]
+        barcodes_as_genes = ["cell0", "cell1", "cell3", "cell4"]
+        with self.assertRaises(ValueError) as cm:
+            MTXIngestor.check_bundle(
+                genes_as_barcodes, barcodes_as_genes, mtx_dimensions
+            )
+        expected_msg = (
+            f"Uploaded matrix suggests {len(genes_as_barcodes)} cells and {len(barcodes_as_genes)} genes "
+            f"instead of {expected_barcodes} cells and {expected_genes} genes. "
+            f"Please transpose your sparse matrix and re-upload. "
+            f"Transposition is expected for matrices exported from AnnData objects. "
```

Comment:
If the other change seems worthwhile:

```suggestion
            f"Uploaded matrix has {len(genes_as_barcodes)} cells and {len(barcodes_as_genes)} genes "
            f"instead of {expected_barcodes} cells and {expected_genes} genes. "
            f"Please transpose your sparse matrix and re-upload. "
            f"Matrices exported from AnnData objects must be transposed."
```

Reply:
see comment above regarding changes, `6ba62e0` & `71b083a`.